code/headers/lidar.hpp

```diff
@@ -0,0 +1,110 @@
+#include <lidar_packets.hpp>
+#include <usart_connection.hpp>
+
+namespace r2d2 {
+    namespace measuring_distance {
+        /**
+         * \brief This class makes it easy to read and extract data from the
+         * lidar using uart.
+         */
+        class lidar_c {
+        public:
+            lidar_packet_header_s header; // This is the fixed amount of data
+                                          // every packet starts with.
+            measurement_data_s measurements[720]; // 720 = 360 * 2. So we sample
```

Comment:
Do you need to sample every half degree? This is currently 720 * 3 = 2160 bytes, which is quite a lot for a distance sensor.